rest_framework/schemas/openapi.py

```diff
@@ -32,39 +32,30 @@ def get_info(self):
 
         return info
 
-    def get_paths(self, request=None):
-        result = {}
-
-        paths, view_endpoints = self._get_paths_and_endpoints(request)
-
-        # Only generate the path prefix for paths that will be included
-        if not paths:
-            return None
+    def get_schema(self, request=None, public=False):
+        """
+        Generate a OpenAPI schema.
+        """
+        self._initialise_endpoints()
 
+        # Iterate endpoints generating per method path operations.
+        # TODO: …and reference components.
+        paths = {}
+        _, view_endpoints = self._get_paths_and_endpoints(None if public else request)
         for path, method, view in view_endpoints:
             if not self.has_view_permissions(path, method, view):
                 continue
+
```

Comment:
Here I intend for per `path, method, view` component generation, components (if any) then becoming parameters to `get_operation()`